packages/core/src/createConfig.ts

```diff
@@ -125,7 +125,10 @@ export function createConfig<
     // If the target chain is not configured, use the client of the current chain.
     type Return = Client<Transport, Extract<chains[number], { id: chainId }>>
     {
+      // chainId specified and not configured
+      if (config.chainId && !chain) throw new ChainNotConfiguredError()
       const client = clients.get(store.getState().chainId)
+      // client exists, but no chain
       if (client && !chain) return client as Return
```

Comment:
This case is extremely unlikely to happen now since it would require `store.getState().chainId` to change from line `122` to `130` so might be worth removing this.